### PR TITLE
[TheiaCoV/TheiaProk/TheiaMeta/TheiaEuk/Freyja_FASTQ] `fastq-scan` updates & improvements. Adding JSON as wf output file

### LINKED ISSUE #471: [feature request] output fastq-scan JSONs in all workflows that include it
:cool:

### :pushpin: Explain the Request

This is a request from me, but everyone will benefit from this add.

I frequently look at the `fastq-scan` output JSON files when reviewing raw & cleaned read lengths and quality scores, and always have to dig into the execution directories for samples individually and it's very annoying to have to click around so much to review these QC metrics.

Can we please output the `fastq-scan` JSON files as part of:

- read_qc_trim_pe subworkflow 
- read_qc_trim_se subworkflow
- TheiaCoV_illumina_pe workflow
- theiacov_illumina_se workflow
- TheiaProk_illumina_pe workflow
- TheiaProk_Illumina_se workflow
- theiameta_illumina_pe workflow
- any other workflows I have missed?

We output the HTML files from fastqc (not the default FASTQ-scanning tool), why not do something similar for fastq-scan? Is there a reason we do not output this file?

### :chart_with_upwards_trend: Desired Behavior

output these JSON files by default, like we do with fastqc html output files

### LINKED ISSUE #571: Move fastq_scan and download_terra_table tasks to use GAR instead of Quay
:cool:

### :pushpin: Explain the Request

fastq_scan uses quay.io 
download_terra_table uses quay.io

Resolve!!!

### PR DESCRIPTION
**Pull request by kapsakcj**
~~Will update this message later~~

This PR closes #471 and closes #571

🗑️ This dev branch should be deleted after merging to main.

## :brain: Summary

This PR updates 2 frequently used tasks (PE and SE versions of the task) `tasks/quality_control/basic_statistics/task_fastq_scan.wdl` and all workflows that use this task to run `fastq-scan`

I mainly did this PR so that the fastq-scan JSON output files are output at the workflow level, but improved other aspects along the way

## :zap: Impacted Workflows/Tasks

- `tasks/quality_control/basic_statistics/task_fastq_scan.wdl`
  - added `set -euo pipefail` in case an error is thrown
  - reduced cpu to 1 as fastq-scan likely cannot utilize multiple threads (AFAIK)
  - reduced disk_size to 50 (don't need much space unless FASTQs are huge)
  - enabled preemptible VMs to reduce cost since it doesn't take very long to run (usually less than 4 minutes)
  - upgraded to fastq-scan v1.0.1 from v0.4.4
  - copied biocontainer docker image to the Theiagen GAR so we don't rely upon quay-hosted biocontainer docker image
  - removed DATE variable and pipeline_date output string as it's unnecessary
  - improved and cleaned up logging and what is printed to STDOUT so it's more clear to user and support team
  - renamed output variable name from `read1_fastq_scan_report` to `read1_fastq_scan_json`
- workflows with 2 or 4 new JSON output files from `fastq-scan`, depending on paired end or single end data
  - `workflows/freyja/wf_freyja_fastq.wdl`
  - `workflows/theiacov/wf_theiacov_clearlabs.wdl`
  - `workflows/theiacov/wf_theiacov_illumina_pe.wdl`
  - `workflows/theiacov/wf_theiacov_illumina_se.wdl`
  - `workflows/theiaeuk/wf_theiaeuk_illumina_pe.wdl`
  - `workflows/theiameta/wf_theiameta_illumina_pe.wdl`
  - `workflows/theiaprok/wf_theiaprok_illumina_pe.wdl`
  - `workflows/theiaprok/wf_theiaprok_illumina_se.wdl`
  - `workflows/utilities/wf_read_QC_trim_pe.wdl`
  - `workflows/utilities/wf_read_QC_trim_se.wdl`
- Other minor and somewhat unrelated changes:
  - updated CI conda environment to not include `defaults` channel due to Anaconda changing their ToS so you have to pay if you use that channel when your organization includes 200+ employees. AFAIK the CI environment does not use any packages from that channel, but better to be safe than sorry. CI runs totally fine without the channel.
  - `workflows/utilities/wf_read_QC_trim_ont.wdl` (no new output files, just corrected the `meta: description` section)'
  - fixed a bug in export_taxon_tables task that was causing silent error, described here: https://github.com/theiagen/public_health_bioinformatics/pull/666#issuecomment-2455412182

This PR may lead to different results in pre-existing outputs: **No**

This PR uses an element that could cause duplicate runs to have different results: **No**

### :gear: Algorithm

upgraded to v1.0.1 of fastq-scan which has support for "large" FASTQs. Not entirely sure what that means other than a slight change in the cpp code, but it's more robust

### ➡️ Inputs

listed above. Inputs that changed are all runtime related

### ⬅️ Outputs

new JSON outputs for read1 and read2, both raw FASTQs and "cleaned" FASTQs to all subworkflows (read_QC subwfs) and workflows that use this task

## :test_tube: Testing

- Freyja_FASTQ
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/b185676e-6d2f-4c44-baa3-cd7fc74280f3)
  - ✅ WF ran successfully, 4 new JSON outputs appear in data table.
- theiacov_clearlabs
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/81b4d1db-cac5-4877-bacb-083b7420848e0)
  - ✅ WF ran successfully, 2 new JSON outputs appear in data table.
- theiacov_illumina_pe
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/cafc93db-b698-4be8-bce9-ac45c951122e)
  - ✅ WF ran successfully, 4 new JSON outputs appear in data table.
- theiacov_illumina_se
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/bf8ccd8a-e7ba-4a98-82c3-6334a280b19f)
  - ✅ WF ran successfully, 2 new JSON outputs appear in data table.
- theiameta_illumina_pe
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/d0cf4936-e429-42e6-94bf-6e76772a5e6e)
  - ✅ WF ran successfully, 4 new JSON outputs appear in data table.
- theiaprok_illumina_pe with export_taxon_tables enabled
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/8cd1d8d2-fc49-4b44-805b-05e05ff014c2)
  - ✅ WF ran successfully, 4 new JSON outputs appear in data table. Additionally, these 2 outputs showed up in the exported taxon table (ecoli_specimen and salmonella_specimen data tables)
- theiaprok_illumina_se with export_taxon_tables enabled
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/79c59a6c-98a9-4702-8569-5795823afd99)
  - ✅ WF ran successfully, 2 new JSON outputs appear in data table. Additionally, these 2 outputs showed up in the exported taxon table (ecoli_specimen_se and salmonella_specimen_se data tables)
- TheiaEuk_illumina_pe
  - [terra wf test](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/17a7ee6c-ec0b-4f4c-9863-94cbdbc4d456)
  - ✅ WF ran successfully, 4 new JSON outputs appear in data table.

### Suggested Scenarios for Reviewer to Test

Would be good to test any of the impacted workflows. No need to test export_taxon_table functionality as I've already done so for both TheiaProk wfs.

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [x] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [x] The documentation has been updated

### CODE REVIEW

File: tests/config/environment.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@ name: pytest-env-CI
 channels:
   - conda-forge
   - bioconda
-  - defaults
```
Comment (kapsakcj) [inline tests/config/environment.yml:5]: We don't have more than 200 people in our organization, but most folks are moving away from using the `defaults` channel due to Anaconda updating their ToS: https://www.anaconda.com/blog/is-conda-free

**Comment by kapsakcj** [comment]
reminder to add the new outputs to theiameta wf as well

**Comment by kapsakcj** [comment]
reminder to add fastq-scan to export_taxon_tables in theiaprok SE wf

**Comment by AndrewLangvt** [comment]
@kapsakcj I know this is still in draft form, but what do you think about outputting the fastq-scan JSONs as an array? That way they only add a single column to Terra table (instead of up to 4), but also allows you to easily access them for support troubleshooting?

**Comment by kapsakcj** [comment]
moving back to to draft state since CI is failing for some unknown reason and since we're considering changing output to an array instead

**Comment by AndrewLangvt** [comment]
After some consideration with the team, we will leave these as separate outputs, not an Array. We will engage the "output consolidation" aspect when we implement TheiaCoV/TheiaProk/etc. "light" versions.

File: tests/config/environment.yml (hunk 1 above)
Comment (AndrewLangvt) [reply tests/config/environment.yml:5]: Thanks for picking up on this & making the quick fix!

**Review by AndrewLangvt — APPROVED** [review]
All tests successful
[Freyja_FASTQ](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/9e84e4cd-322c-4108-8bcb-0d7472dd0b32)
[theiacov_clearlabs](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/a84bc60d-f349-4c17-bd96-2c15a789d6bd)
[theiacov_illumina_pe](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/55ead88c-6b6d-4acf-bb9e-e330a67a100f)
[theiacov_illumina_se](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/2e591d06-c78e-4630-b243-6c8155f0b6fc)
[theiameta_illumina_pe](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/c8bde8b4-1147-4684-8f02-7c74f946247f)
[theiaprok_illumina_pe](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/feff9ae3-1e10-47f6-9258-ff8f725887dd)
[theiaeuk_illumina_pe](https://app.terra.bio/#workspaces/theiagen-nepgcoe/NEPGCOE_Bioinformatics_Development/job_history/51401a9a-5c08-4a98-8b64-ab1f499c6a35)

Clean, simple updates! Thanks @kapsakcj

**Pull request merged**